### PR TITLE
TweetContentProvider to show inline tweets

### PR DESCRIPTION
**Pull request by DanTup**
Added TweetContentProvider to show inline tweets when Twitter URLs are pasted in chat.

### CODE REVIEW

File: Chat/Chat.cs [hunk 1]
```diff
@@ -26,7 +26,8 @@ namespace SignalR.Samples.Hubs.Chat {
         private static readonly List<IContentProvider> _contentProviders = new List<IContentProvider>() {
             new ImageContentProvider(),
             new YouTubeContentProvider(),
-            new CollegeHumorContentProvider()
+            new CollegeHumorContentProvider(),
+			new TweetContentProvider()
```
Comment (davidfowl) [inline Chat/Chat.cs:30]: Formatting :)

File: Chat/ContentProviders/TweetContentProvider.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+using System.Web;
+
+namespace SignalR.Samples.Hubs.Chat.ContentProviders
+{
+	/// <summary>
+	/// Content Provider that provides the necessary functionality to show Tweets in the chat when someone pastes a
+	/// Twitter link (eg. http://twitter.com/#!/DanTup/status/117940664771162112). Send feedback/issues to Danny Tuppeny
+	/// (@DanTup on Twitter).
+	/// </summary>
+	public class TweetContentProvider : IContentProvider
```
Comment (davidfowl) [inline Chat/ContentProviders/TweetContentProvider.cs:14]: Brace style

File: Chat/Chat.cs (hunk 1 above)
Comment (DanTup) [inline Chat/Chat.cs:30]: This lines up if your tabstops are the right number of spaces ;-P

Comment (davidfowl) [inline Chat/Chat.cs:30]: Wait you used tabs?

Comment (DanTup) [inline Chat/Chat.cs:30]: Er, no...

File: Chat/ContentProviders/TweetContentProvider.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		/// </summary>
+		private static readonly Regex tweetRegex = new Regex(@".*http://twitter.com/(?:#!/)*[^/]+/status/(\d+).*");
+
+		/// <summary>
+		/// The block of HTML/Script that is sent to the client to render the tweet text. The tweet ID will be passed
+		/// in as {0}.
+		/// </summary>
+		private static readonly string tweetScript = string.Format( // Be aware: Nested string.format placeholder!
+			"<div class=\"tweet_{{0}}\"><script src=\"{0}\"></script></div>",
+			HttpUtility.HtmlEncode("http://api.twitter.com/1/statuses/show/{0}.json?include_entities=false&callback=addTweet")
```
Comment (davidfowl) [inline Chat/ContentProviders/TweetContentProvider.cs:27]: JsonP eh?

Comment (DanTup) [inline Chat/ContentProviders/TweetContentProvider.cs:27]: eh?

Comment (davidfowl) [inline Chat/ContentProviders/TweetContentProvider.cs:27]: Looks good. Don't worry I'll fix the formatting issues :)

Comment (DanTup) [inline Chat/ContentProviders/TweetContentProvider.cs:27]: Already doe... incoming :P

Comment (DanTup) [inline Chat/ContentProviders/TweetContentProvider.cs:27]: *done

**Pull request merged**